src/sysctl/sysctl.c

```diff
@@ -83,24 +154,29 @@ static int apply_all(OrderedHashmap *sysctl_options) {
         ORDERED_HASHMAP_FOREACH(option, sysctl_options, i) {
                 int k;
 
-                k = sysctl_write(option->key, option->value);
-                if (k < 0) {
-                        /* If the sysctl is not available in the kernel or we are running with reduced
-                         * privileges and cannot write it, then log about the issue, and proceed without
-                         * failing. (EROFS is treated as a permission problem here, since that's how
-                         * container managers usually protected their sysctls.) In all other cases log an
-                         * error and make the tool fail. */
-
-                        if (option->ignore_failure || k == -EROFS || ERRNO_IS_PRIVILEGE(k))
-                                log_debug_errno(k, "Couldn't write '%s' to '%s', ignoring: %m", option->value, option->key);
-                        else if (k == -ENOENT)
-                                log_info_errno(k, "Couldn't write '%s' to '%s', ignoring: %m", option->value, option->key);
-                        else {
-                                log_error_errno(k, "Couldn't write '%s' to '%s': %m", option->value, option->key);
-                                if (r == 0)
-                                        r = k;
-                        }
-                }
+                k = sysctl_write_or_warn(option->key, option->value, option->ignore_failure);
+                if (r == 0)
+                        r = k;
+
+                /* "Propagate" net.*.conf.default settings to interfaces which were already created before we
+                 * managed to write the default value. This is done to close the race between network
+                 * interface probing and systemd-sysctl.service running at boot.
+                 *
+                 * The keys use slashes as separators and are normalized, see sysctl_normalize above. */
+                _cleanup_free_ char *prefix = NULL, *rest = NULL;
+                int n;
+
+                k = sscanf(option->key, "net/%*[^/]/conf%n/default/%ms", &n, &rest);
```

Comment:
i must say I am not fond of new sscanf() expressions in our code (space handling is just so weird)... In particular as this looks like something easily parsed without with path_startswith() and friends